### PR TITLE
[Execution] check executed result is consistent with the sealed result

### PR DESCRIPTION
**Pull request by zhangchiqing**
In order to prevent an EN from keeping executing blocks on a wrong execution fork, this PR added checks to compare the executed result with sealed result.

The check has been added to two places:

when a block is finalized
when a block is executed
The reason the above places are needed is that a block might be sealed but not executed, or executed but not sealed.

### CODE REVIEW

File: engine/execution/checker/engine.go [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+}
+
+func (e *Engine) SubmitLocal(event interface{})                      {}
+func (e *Engine) Submit(originID flow.Identifier, event interface{}) {}
+func (e *Engine) ProcessLocal(event interface{}) error {
+	return fmt.Errorf("check don't process message")
+}
+func (e *Engine) Process(originID flow.Identifier, event interface{}) error {
+	return fmt.Errorf("check don't process message")
+}
```
Comment (jordanschalm) [inline engine/execution/checker/engine.go:60]: Do we need all this?

**Review by jordanschalm — APPROVED** [review]

File: engine/execution/checker/engine.go [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+		e.log.Fatal().Err(err).Msg("execution consistency check failed")
+	}
+}
+
+// CheckExecutionConsistencyWithBlock when a block is executed, if the block has been sealed, check whether if the sealed result is consistent
+// with the executed result
+func (e *Engine) CheckExecutionConsistencyWithBlock(block *flow.Block, mycommit flow.StateCommitment) {
+	err := e.checkExecutedBlockConsistency(block, mycommit)
+	if err != nil {
+		e.log.Fatal().Err(err).
```
Comment (m4ksio) [inline engine/execution/checker/engine.go:77]: Won't crashing a node casue an infinite loop? Container will restart and we are back to the same situation.
Shouldn't we rather have some idle-mode, so keep the node alive, but doing essentially nothing?

Comment (zhangchiqing) [reply engine/execution/checker/engine.go:68]: Good point. 

However, I was afraid the alive node is still able to respond to receipt requests from consensus.

**Comment by zhangchiqing** [comment]
Close in favor of https://github.com/onflow/flow-go/pull/441

**Pull request closed without merging**